File: org.summer.view.widget/src/org/summer/view/widget/data/InterfaceConverter.java
package org.summer.view.widget.data;
/*internal*/ public class InterfaceConverter implements IValueConverter
{ 
    //----------------------------------------------------- 
    //
    //  Constructors 
    //
    //------------------------------------------------------

    /*internal*/ public InterfaceConverter(Type sourceType, Type targetType) 
    {
        _sourceType = sourceType; 
        _targetType = targetType; 
    }

    //-----------------------------------------------------
    //
    //  Interfaces (IValueConverter)
    // 
    //------------------------------------------------------

    public Object Convert(Object o, Type type, Object parameter, CultureInfo culture) 
    {
        return ConvertTo(o, _targetType); 
    }

    public Object ConvertBack(Object o, Type type, Object parameter, CultureInfo culture)
    { 
        return ConvertTo(o, _sourceType);
    } 

    private Object ConvertTo(Object o, Type type)
    { 
        return type.IsInstanceOfType(o) ? o : null;
    }

    Type _sourceType; 
    Type _targetType;
}

File: summer/WebContent/js/MS/internal/data/InterfaceConverter.js
/**
 * InterfaceConverter
 */

define(["dojo/_base/declare", "system/Type", "data/IValueConverter"], 
		function(declare, Type, IValueConverter){
	var InterfaceConverter = declare("InterfaceConverter", IValueConverter,{
		constructor:function(/*Type*/ sourceType, /*Type*/ targetType) 
        {
            this._sourceType = sourceType; 
            this._targetType = targetType; 
        },
        
//        public object 
        Convert:function(/*object*/ o, /*Type*/ type, /*object*/ parameter, /*CultureInfo*/ culture) 
        {
            return this.ConvertTo(o, this._targetType); 
        },

//        public object 
        ConvertBack:function(/*object*/ o, /*Type*/ type, /*object*/ parameter, /*CultureInfo*/ culture)
        { 
            return this.ConvertTo(o, this._sourceType);
        }, 
 
//        private object 
        ConvertTo:function(/*object*/ o, /*Type*/ type)
        { 
            return type.IsInstanceOfType(o) ? o : null;
        }
        
	});
	
	Object.defineProperties(InterfaceConverter.prototype,{
		  
	});
	
	Object.defineProperties(InterfaceConverter,{
		  
	});
	
	InterfaceConverter.Type = new Type("InterfaceConverter", InterfaceConverter, [IValueConverter.Type]);
	return InterfaceConverter;
});
